improver_tests/generate_ancillaries/test_GenerateTimezoneMask.py

```diff
@@ -0,0 +1,296 @@
+    assert result.ndim == 1
+    assert isinstance(result, np.ndarray)
+    assert result.dtype == np.int32
+
+    # Southern hemisphere winter
+    time = datetime(2020, 7, 1, 12, tzinfo=pytz.utc)
+    plugin = GenerateTimezoneMask(time=time)
+    result = plugin._calculate_tz_offsets(coordinate_pairs)
+    assert_array_equal(result, expected)
+
+    # Test including daylights savings, so the result should change as the
+    # date is changed.
+
+    # Northern hemisphere winter
+    expected = [-5 * 3600, 0, 11 * 3600]
+    time = datetime(2020, 1, 1, 12, tzinfo=pytz.utc)
+    plugin = GenerateTimezoneMask(ignore_dst=False, time=time)
+    result = plugin._calculate_tz_offsets(coordinate_pairs)
+    assert_array_equal(result, expected)
+
+    # Southern hemisphere winter
+    expected = [-4 * 3600, 1 * 3600, 10 * 3600]
+    time = datetime(2020, 7, 1, 12, tzinfo=pytz.utc)
+    plugin = GenerateTimezoneMask(ignore_dst=False, time=time)
+    result = plugin._calculate_tz_offsets(coordinate_pairs)
+    assert_array_equal(result, expected)
+
+
+@pytest.mark.parametrize("grid_fixture", ["global_grid", "uk_grid"])
+def test__create_template_cube(request, grid_fixture):
+    """Test the construction of a template cube slice, checking the shape
+    data types, and attributes."""
+
+    grid = request.getfixturevalue(grid_fixture)
+    time = datetime(2020, 1, 1, 12, tzinfo=pytz.utc)
+
+    expected = {
+        "global_grid": {"shape": (19, 37), "attributes": GLOBAL_ATTRIBUTES},
+        "uk_grid": {"shape": (21, 22), "attributes": UK_ATTRIBUTES},
+    }
+
+    for ignore_dst in [True, False]:
+
+        # Set expected includes_daylights_savings attribute
+        expected[grid_fixture]["attributes"]["includes_daylights_savings"] = str(
+            not ignore_dst
+        )
+
+        plugin = GenerateTimezoneMask(ignore_dst=ignore_dst, time=time)
+        result = plugin._create_template_cube(grid)
+
+        assert result.name() == "timezone_mask"
+        assert result.units == 1
+        assert result.coord("time").points[0] == time.timestamp()
+        assert result.coord("time").dtype == np.int64
+        assert result.shape == expected[grid_fixture]["shape"]
+        assert result.dtype == np.int32
+        assert result.attributes == expected[grid_fixture]["attributes"]
+
+
+def test__group_timezones(timezone_mask):
+    """Test the grouping of different UTC offsets into larger groups using a
+    user provided specification. The input cube list contains cubes corresponding
+    to 4 UTC offsets. Two tests are run, grouping these first into equal sized
+    groups, and then into unequally sized groups."""
+
+    groupings = [{0: [0, 1], 1: [2, 3]}, {0: [0, 2], 1: [3]}]
+
+    for groups in groupings:
+        plugin = GenerateTimezoneMask(groupings=groups)
+        result = plugin._group_timezones(timezone_mask)
+
+        assert len(result) == len(groups)
+        for group, cube in zip(groups.values(), result):
```

Comment:
Codacy found an issue: [dict.values referenced when not iterating](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5879946)